src/screens/NetworkSettings.js

```diff
@@ -0,0 +1,216 @@
+import ReactLoading from 'react-loading';
+import hathorLib from '@hathor/wallet-lib';
+import {
+  DEFAULT_SERVERS,
+  DEFAULT_WALLET_SERVICE_SERVERS,
+  DEFAULT_WALLET_SERVICE_WS_SERVERS,
+  NETWORK_SETTINGS,
+  NETWORK_SETTINGS_STATUS,
+  colors
+} from '../constants';
+import { useDispatch, useSelector } from 'react-redux';
+import { GlobalModalContext, MODAL_TYPES } from '../components/GlobalModal';
+import LOCAL_STORE from '../storage';
+import { getGlobalWallet } from "../modules/wallet";
+import { useNavigate } from 'react-router-dom';
+import { networkSettingsRequestUpdate, setNetworkSettingsStatus } from "../actions";
+import BackButton from '../components/BackButton';
+
+
+/**
+ * Screen to change the settings of the network the wallet is connected to.
+ *
+ * @memberof Screens
+ */
+function NetworkSettings() {
+  const context = useContext(GlobalModalContext);
+  const navigate = useNavigate();
+  const dispatch  = useDispatch();
+
+  const networkSelectRef = useRef(null);
+  const alertSuccessRef = useRef(null);
+  const pinRef = useRef(null);
+  const formRef = useRef(null);
+  const nodeRef = useRef(null);
+  const txMiningRef = useRef(null);
+  const explorerRef = useRef(null);
+  const explorerServiceRef = useRef(null);
+  const walletServiceRef = useRef(null);
+  const walletServiceWSRef = useRef(null);
+
+  const wallet = getGlobalWallet();
+
+  // Use selector to fetch current network settings
+  const { networkSettings } = useSelector(state => ({
+    networkSettings: state.networkSettings
+  }));
+
+  /* loading {boolean} If should show spinner while waiting for server response */
+  const [loading, setLoading] = useState(false);
+  /* errorMessage {string} Message to be shown in case of error in form */
+  const [errorMessage, setErrorMessage] = useState('');
+  /* networkSelected {string} Network that the user wants to connect */
+  const [networkSelected, setNetworkSelected] = useState('');
+  /* isHardwareWallet {boolean} If the application is using a hardware wallet */
+  const [isHardwareWallet] = useState(LOCAL_STORE.isHardwareWallet());
+
+  useEffect(() => {
+    if (networkSettings.status === NETWORK_SETTINGS_STATUS.WAITING_NETWORK_CONFIRMATION) {
+      // User requested to change to a node that is a testnet or privatenet
+      // so we must show the confirmation modal before changing the network
+      context.showModal(MODAL_TYPES.CONFIRM_TESTNET, {
+        success: () => onNetworkConfirmed(),
+        onUserCancel: () => onNetworkConfirmationCancel(),
+        network: networkSettings.newNetwork,
+      });
+      setLoading(false);
+    }
+
+    if (networkSettings.status === NETWORK_SETTINGS_STATUS.LOADING) {
+      setLoading(true);
+    }
+
+    if (networkSettings.status === NETWORK_SETTINGS_STATUS.ERROR) {
+      setErrorMessage(networkSettings.error)
+      setLoading(false);
+    }
+
+    if (networkSettings.status === NETWORK_SETTINGS_STATUS.READY) {
+      setLoading(false);
```

Comment:
Do you really need to set state here?

Reply:
Removed all unecessary component states https://github.com/HathorNetwork/hathor-wallet/pull/701/commits/4fc677fbd7aba40d084447ce3314e4b6f4a97ae1